plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/function/tablechanges/DeltaLakeTableChangesFunction.java

```diff
@@ -0,0 +1,190 @@
+        }
+
+        ScalarArgument startVersionArgument = (ScalarArgument) arguments.get(START_VERSION_ARGUMENT);
+        Long startVersion = (Long) startVersionArgument.getValue();
+        requireNonNull(startVersion, "startVersion value for function table_changes() is null");
+        startVersion++; // to guarantee exclusiveness for the lower end of the range
+        checkArgument(startVersion >= 0, "startVersion value for function table_changes() must not be negative");
+
+        ScalarArgument endVersionArgument = (ScalarArgument) arguments.get(END_VERSION_ARGUMENT);
+        Long endVersion = (Long) endVersionArgument.getValue();
+        requireNonNull(endVersion, "endVersion value for function table_changes() is null");
+        checkArgument(endVersion >= 0, "endVersion value for function table_changes() must not be negative");
+        checkArgument(endVersion >= startVersion, "endVersion value for function table_changes() must be greater than startVersion");
+
+        int indexOfDot = schemaTableName.indexOf(".");
+        String schemaName = schemaTableName.substring(0, indexOfDot);
+        String tableName = schemaTableName.substring(indexOfDot + 1);
+        MetadataEntry metadata = getTableMetadata(session, schemaName, tableName);
+
+        ImmutableList.Builder<Descriptor.Field> outputFields = ImmutableList.builder();
+        List<DeltaLakeColumnMetadata> deltaLakeColumnMetadata = extractSchema(metadata, typeManager);
+        deltaLakeColumnMetadata.stream()
+                .map(column -> new Descriptor.Field(column.getName(), Optional.of(column.getType())))
+                .forEach(outputFields::add);
+        outputFields.add(new Descriptor.Field(CHANGE_TYPE_COLUMN_NAME, Optional.of(VARCHAR)));
+        outputFields.add(new Descriptor.Field(COMMIT_VERSION_COLUMN_NAME, Optional.of(INTEGER)));
+        outputFields.add(new Descriptor.Field(COMMIT_TIMESTAMP_COLUMN_NAME, Optional.of(TIMESTAMP_TZ_MILLIS)));
+        List<Descriptor.Field> requiredColumns = outputFields.build();
+
+        List<DeltaLakeColumnHandle> columns = deltaLakeColumnMetadata.stream()
+                .map(column -> toColumnHandle(
+                        column.getColumnMetadata(),
+                        column.getFieldId(),
+                        column.getPhysicalName(),
+                        column.getPhysicalColumnType(),
+                        metadata.getCanonicalPartitionColumns()))
+                .collect(toImmutableList());
+
+        return TableFunctionAnalysis.builder()
+                .handle(new TableChangesFunctionTableHandle(catalogHandle, schemaName, tableName, startVersion, endVersion, columns))
+                .returnedType(new Descriptor(requiredColumns))
+                .build();
+    }
+
+    private MetadataEntry getTableMetadata(ConnectorSession session, String schemaName, String tableName)
+    {
+        HiveMetastoreBackedDeltaLakeMetastore metastore = new HiveMetastoreBackedDeltaLakeMetastore(
```

Comment:
Inject a `DeltaLakeMetadataFactory` instead of using `HiveMetastoreBackedDeltaLakeMetastore` directly